lib/compilers/dotnet.ts

```diff
@@ -30,28 +30,28 @@ import fs from 'fs-extra';
 import { BaseCompiler } from '../base-compiler';
 
 class DotNetCompiler extends BaseCompiler {
-    private rID: string;
-    private targetFramework: string;
-    private buildConfig: string;
-    private nugetPackagesPath: string;
-    private clrBuildDir: string;
-    private additionalSources: string;
-    private langVersion: string;
+    private coreRoot: string;
+    private crossgen2bin: string;
+    private testAppSrc: string;
+    private testAppName: string;
+    private testAppOutputDir: string;
 
     constructor(compilerInfo, env) {
         super(compilerInfo, env);
 
-        this.rID = this.compilerProps(`compiler.${this.compiler.id}.runtimeId`);
-        this.targetFramework = this.compilerProps(`compiler.${this.compiler.id}.targetFramework`);
-        this.buildConfig = this.compilerProps(`compiler.${this.compiler.id}.buildConfig`);
-        this.nugetPackagesPath = this.compilerProps(`compiler.${this.compiler.id}.nugetPackages`);
-        this.clrBuildDir = this.compilerProps(`compiler.${this.compiler.id}.clrDir`);
-        this.additionalSources = this.compilerProps(`compiler.${this.compiler.id}.additionalSources`);
-        this.langVersion = this.compilerProps(`compiler.${this.compiler.id}.langVersion`);
+        this.coreRoot = this.compilerProps(`compiler.${this.compiler.id}.coreRoot`);
+        this.testAppSrc = this.compilerProps(`compiler.${this.compiler.id}.testAppSrc`);
+        this.testAppName = path.basename(this.testAppSrc);
+        this.crossgen2bin = path.join(this.coreRoot, 'crossgen2', 'crossgen2.dll');
+        this.testAppOutputDir = 'out';
     }
 
     get compilerOptions() {
-        return ['publish', '-c', this.buildConfig, '--self-contained', '--runtime', this.rID, '-v', 'q', '--nologo'];
+        return ['build', this.testAppName + '.csproj', '-c', 'Release', '-o', this.testAppOutputDir, 
+        // Properties:
+        '/p:AllowUnsafeBlocks=true',
```

Comment:
does this change cause any compiler differences? We try our best to make it so pre-existing links remain valid. e.g. in a C++ context we wouldn't change the compiler flags to add or remove things that affect code generation.

Reply:
I think we can move this one to the pregenerated template just fine.

This property allows users to play with raw pointers in C#

Reply:
So these don't affect codegen at all so I guess it's fine to keep them here?
AllowUnsafeBlocks should always be enabled, without it it's not possible to compile a simple raw pointer like `int* p = &b;`
Nullable is disabled just to hide some boring garbage from the output nobody will ever look at, e.g. see this: https://godbolt.org/z/xfh734ejs

it's expected to be just `ret` but we're seeing stuff generated by Roslyn for Nullability analysis (Attribute definition)